From bf8487d469a846a802be969778e4e1efc67e2624 Mon Sep 17 00:00:00 2001
Message-Id: <bf8487d469a846a802be969778e4e1efc67e2624.1423172493.git.feitong.yi@intel.com>
In-Reply-To: <18bb7324f916bffbc75e07ea0ffd3e6e68e7b6d8.1423172493.git.feitong.yi@intel.com>
References: <18bb7324f916bffbc75e07ea0ffd3e6e68e7b6d8.1423172493.git.feitong.yi@intel.com>
From: Gaurav K Singh <gaurav.k.singh@intel.com>
Date: Thu, 15 Jan 2015 01:16:08 +0530
Subject: [PATCH 28/29] MUST_REBASE [VPG]: video/adf/intel: Use cck reg to
 check DSI Pll status

Instead of pipe configuration reg, use cck reg for checking whether
DSI Pll is getting locked.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed

Issue: GMINL-4868
Change-Id: Ia7750a1a130f99b759a4fa00a7ca336fc6f1e4ca
Signed-off-by: Gaurav K Singh <gaurav.k.singh@intel.com>
---
 drivers/video/adf/intel/core/vlv/vlv_pipe.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/video/adf/intel/core/vlv/vlv_pipe.c b/drivers/video/adf/intel/core/vlv/vlv_pipe.c
index 9442817..7845477 100644
--- a/drivers/video/adf/intel/core/vlv/vlv_pipe.c
+++ b/drivers/video/adf/intel/core/vlv/vlv_pipe.c
@@ -238,7 +238,8 @@ u32 vlv_pipe_disable(struct vlv_pipe *pipe)
 bool vlv_pipe_wait_for_pll_lock(struct vlv_pipe *pipe)
 {
 	u32 err = 0;
-	if (wait_for(REG_READ(pipe->offset) & PIPECONF_DSI_PLL_LOCKED, 20)) {
+	if (wait_for(vlv_cck_read(CCK_REG_DSI_PLL_CONTROL) &
+		     DSI_PLL_LOCK, 20)) {
 		pr_err("DSI PLL lock failed\n");
 		err = -EINVAL; /* FIXME: assign correct error */
 	}
-- 
1.7.9.5

